### DISCUSSION TITLE
Remove baseUrl from tsconfig.json in all stacks

### DISCUSSION THREAD

**Discussion by silvenon** [remix-run/remix#6997] [Proposals]
Configuring `baseUrl` was a source of confusion for me, but I assume that it's there because of `paths`, however it's not needed for that:

> As of TypeScript 4.1, `baseUrl` is no longer required to be set when using `paths`.
>
> https://www.typescriptlang.org/tsconfig#baseUrl

My source of confusion was here:

> This resolution has higher priority than lookups from `node_modules`.

This is not true if `moduleResolution` is something like `node` or `bundler`. Probably because:

> This feature was designed for use in conjunction with AMD module loaders in the browser, and is not recommended in any other context.

If it's not recommended in our context, what do you think about removing it?

**Comment by MichaelDeBoey** [comment]
This will be fixed by #7008

**Reply by MichaelDeBoey** [reply]
@silvenon It appears that we need `baseUrl` for some reason and we're throwing an error if it's not present

https://github.com/remix-run/remix/blob/d45f71febf1e1c9000b35fb72a48ea20614329da/packages/remix-dev/compiler/utils/tsconfig.ts#L15-L19
